main/src/ui/util/helper.vala

```diff
@@ -144,59 +144,87 @@ public static bool is_24h_format() {
+
+        s = url_regex.replace_eval(s, -1, 0, 0, (mi, s) => {
+            string url = mi.fetch(0);
+
+            s.append_printf("<a href=\"%s\">%s</a>", Markup.escape_text(url), url);
+            return false;
+        });
+
+        // Now find all the URLs without replacing anything
+        s = url_regex.replace_eval(s, -1, 0, 0, (mi, s) => {
             int start, end;
-            match_info.fetch_pos(0, out start, out end);
-            string link = s[start:end];
-            return parse_add_markup(s[0:start], highlight_word, parse_links, parse_text_markup, already_escaped) +
-                "<a href=\"" + Markup.escape_text(link) + "\">" + parse_add_markup(link, highlight_word, false, false, already_escaped) + "</a>" +
-                parse_add_markup(s[end:s.length], highlight_word, parse_links, parse_text_markup, already_escaped);
-        }
-    }
+            mi.fetch_pos(0, out start, out end);
+            start_url_pos.add(start);
+            end_url_pos.add(end);
+
+            s.append_printf("%s", mi.fetch(0));
 
-    if (!already_escaped) {
-        s = Markup.escape_text(s);
-        already_escaped = true;
+            return false;
+        });
     }
 
     if (highlight_word != null) {
         Regex highlight_regex = new Regex("\\b" + Regex.escape_string(highlight_word.down()) + "\\b");
-        MatchInfo match_info;
-        highlight_regex.match(s.down(), 0, out match_info);
-        if (match_info.matches()) {
+
+        s = highlight_regex.replace_eval(s, -1, 0, 0, (mi, s) => {
             int start, end;
-            match_info.fetch_pos(0, out start, out end);
-            return parse_add_markup(s[0:start], highlight_word, parse_links, parse_text_markup, already_escaped) +
-                "<b>" + s[start:end] + "</b>" +
-                parse_add_markup(s[end:s.length], highlight_word, parse_links, parse_text_markup, already_escaped);
-        }
+            mi.fetch_pos(0, out start, out end);
+
+
+            for (int x = 0; x < start_url_pos.size; x++) {
+                // Check to see if this match starts within an URL
+                if (start > start_url_pos.get(x) && start < end_url_pos.get(x)) {
+                    s.append_printf("%s", mi.fetch(0));
+                    return false;
+                }
+                // Check to see if this match ends within an URL
+                if (end > start_url_pos.get(x) && end < end_url_pos.get(x)) {
+                    s.append_printf("%s", mi.fetch(0));
+                    return false;
+                }
+            }
+            s.append_printf("<b>%s</b>", mi.fetch(0));
+            return false;
+        });
     }
 
     if (parse_text_markup) {
-        string[] markup_string = new string[]{"`", "_", "*"};
-        string[] convenience_tag = new string[]{"tt", "i", "b"};
+        string[] markup_string = new string[]{"`", "_", "*", "~"};
```

Comment:
So this PR also adds strikethrough support?

Reply:
Yeah, I did it after the initial pull request. Maybe I need to update the PR comment.